tracer/build/_build/Build.Steps.cs

```diff
@@ -613,23 +613,49 @@ async Task DownloadWafVersion(string libddwafVersion = null, string uncompressFo
         .Unlisted()
         .After(CompileManagedSrc)
         .Executes(() =>
-        {
-            var targetFrameworks = IsWin
-                ? TargetFrameworks
-                : TargetFrameworks.Where(framework => !framework.ToString().StartsWith("net4"));
-
-            // Publish Datadog.Trace.MSBuild which includes Datadog.Trace
-            DotNetPublish(s => s
-                .SetProject(Solution.GetProject(Projects.DatadogTraceMsBuild))
-                .SetConfiguration(BuildConfiguration)
-                .SetTargetPlatformAnyCPU()
-                .EnableNoBuild()
-                .EnableNoRestore()
-                .CombineWith(targetFrameworks, (p, framework) => p
-                    .SetFramework(framework)
-                    .SetOutput(MonitoringHomeDirectory / framework)));
-        });
-
+                                        {
+                                            if (PublishReadyToRun)
+                                            {
+                                                var targetFramework = TargetFramework.NET6_0;
+                                                Logger.Information($"ReadyToRun is {PublishReadyToRun}");
+
+                                                // Needed as we need to restore with the RuntimeIdentifier
+                                                DotNetRestore(s => s
+                                                                  .SetProjectFile(Solution.GetProject(Projects.DatadogTraceMsBuild))
+                                                                  .SetPublishReadyToRun(PublishReadyToRun)
+                                                                  .SetRuntime(RuntimeIdentifier)
+                                                );
```

Comment:
Yeah, it's already set in `DotNetPublish`, but for some reason, if this is not set, it will complain and will never compile properly! So a restore is needed with the current `Runtime`